Documentation/flex-to-csi-migration.md

```diff
@@ -0,0 +1,68 @@
+
+**NOTE**: The Migration procedure will come with a downtime as we need to scale down the applications using the volumes before migration.
+
+## Migrate a PVC
+
+1. Stop the application pods that are consuming the flex volume(s) that need to be converted.
+2. Connect to migration pod
+   1. `migration_pod=$(kubectl -n rook-ceph get pod -l app=rook-ceph-migrator -o jsonpath='{.items[*].metadata.name}')`
+
+   2. `kubectl -n rook-ceph exec -it "$migration_pod" -- sh`
+3. Run the tool to migrate the PVC. See the [section below](#migration-tool-options) for more details on the options and the sample output.
+   1. `pv-migrator --pvc=<pvc-name> --pvc-ns=<pvc-namespace> --destination-sc=<storageclass-name-to-migrate-in> --rook-ns=<rook-ceph> --ceph-cluster-ns=<ceph-cluster-namespace>`
+4. Start the application pods which were stopped in step 1.
+
+## Migration Tool Options
+
+These are the options for converting a single PVC. For more options, see the [tool documentation](https://github.com/ceph/persistent-volume-migrator), for example to convert all PVCs automatically that belong to the same storage class.
+
+1. `--pvc`: **required**: reads name of the pvc to migrate
+2. `--pvc-ns`: **required**: reads the namespace of the PVC which is going to migrate
+3. `--destination-sc`: **required**: read the name of the storage class in which you want to migrate.
+4. `--rook-ns`: **optional** namespace where the rook operator is running. **default: rook-ceph**.
+5. `--ceph-cluster-ns`: **optional** namespace where the ceph cluster is running. **default: rook-ceph**.
+
+```console
+pv-migrator --pvc=rbd-pvc --pvc-ns=default --destination-sc=csi-rook-ceph-block
+```
+
+```console
+I1125 07:56:22.247311      63 log.go:34] Create Kubernetes Client
+I1125 07:56:22.259115      63 log.go:34] List all the PVC from the source storageclass
+I1125 07:56:22.261205      63 log.go:34] 1 PVCs found with source StorageClass
+I1125 07:56:22.261221      63 log.go:34] Start Migration of PVCs to CSI
+I1125 07:56:22.261226      63 log.go:34] migrating PVC "rbd-pvc" from namespace "default"
+I1125 07:56:22.261229      63 log.go:34] Fetch PV information from PVC rbd-pvc
+I1125 07:56:22.266734      63 log.go:34] PV found "pvc-30a01887-8821-4baf-835c-16a7e55ba7f0"
+---
+I1125 07:56:26.483172      63 log.go:34] successfully renamed volume csi-vol-2f8de58f-4dc5-11ec-a130-0242ac110005 -> pvc-30a01887-8821-4baf-835c-16a7e55ba7f0
+I1125 07:56:26.483194      63 log.go:34] Delete old PV object: pvc-30a01887-8821-4baf-835c-16a7e55ba7f0
+I1125 07:56:26.504578      63 log.go:34] waiting for PV pvc-30a01887-8821-4baf-835c-16a7e55ba7f0 in state &PersistentVolumeStatus{Phase:Bound,Message:,Reason:,} to be deleted (0 seconds elapsed)
+I1125 07:56:26.702921      63 log.go:34] deleted persistent volume pvc-30a01887-8821-4baf-835c-16a7e55ba7f0
+I1125 07:56:26.702944      63 log.go:34] successfully migrated pvc rbd-pvc
+I1125 07:56:26.703335      63 log.go:34] Successfully migrated all the PVCs to CSI
```

Comment:
If only one PVC is being migrated, the tool should skip printing this line, it was confusing.
```suggestion
```

Reply:
this I'll need to change in tool side first. will open PR

Reply:
We need to log something after successful migration. How about this 
1. `Successfully migrated flex PVCs to CSI`
2. `Migration done successfully`

Reply:
The message for individual PVC names seems sufficient. Or instead of saying "all" PVCs, how about printing the number of PVCs converted?